From: "Fabian P. Schmidt" <kerel-fs@gmx.de>
Date: Thu, 9 Feb 2017 22:23:07 +0100
Subject: Improve Makefile

---
 Makefile | 46 ++++++++++++++++++++++------------------------
 1 file changed, 22 insertions(+), 24 deletions(-)

diff --git a/Makefile b/Makefile
index 1c7b78c..3d53c5f 100644
--- a/Makefile
+++ b/Makefile
@@ -1,41 +1,39 @@
 VERSION = $(shell cat VERSION.txt)
 
-# USE_RPI_GPU_FFT = 1
-
 FLAGS = -Wall -O3 -ffast-math -DVERSION=$(VERSION)
-LIBS  = -lpthread -lm -ljpeg -lconfig -lrt
+LDLIBS  =  -lpthread -lm -ljpeg -lconfig -lrt -lfftw3 -lfftw3f -lrtlsdr -ldl
+
+#####
 
 ifneq ("$(wildcard /opt/vc/src/hello_pi/hello_fft)","")
+TARGET=RASPBERRYPI
+endif
+
+ifeq ($(TARGET),RASPBERRYPI)
 USE_RPI_GPU_FFT = 1
 FLAGS += -mcpu=arm1176jzf-s -mtune=arm1176jzf-s -march=armv6zk -mfpu=vfp
 endif
 
-#ifndef __MACH__ # _POSIX_TIMERS  # OSX has no clock_gettime() and thus not -lrt (but _POSIX_TIMERS seem not to be defined in make ?)
-#LIBS += -lrt
-#endif
-
 ifdef USE_RPI_GPU_FFT
+LDLIBS += -ldl
 GPU_FLAGS = -DUSE_RPI_GPU_FFT
-GPU_SRC   = mailbox.c gpu_fft_base.c gpu_fft.c gpu_fft_twiddles.c gpu_fft_shaders.c
-LIBS += -ldl
+GPU_SRC   = src/mailbox.c src/gpu_fft.c src/gpu_fft_base.c src/gpu_fft_twiddles.c src/gpu_fft_shaders.c
 endif
 
-all:    gsm_scan ogn-rf r2fft_test
+#####
 
-ogn-rf:       Makefile ogn-rf.cc rtlsdr.h thread.h fft.h buffer.h image.h
-	g++ $(FLAGS) $(GPU_FLAGS) -o ogn-rf ogn-rf.cc $(GPU_SRC) $(LIBS) -lrtlsdr -lfftw3 -lfftw3f
-ifdef USE_RPI_GPU_FFT
-	sudo chown root ogn-rf
-	sudo chmod a+s  ogn-rf
-endif
+all: gsm_scan ogn-rf r2fft_test
 
-gsm_scan:       Makefile gsm_scan.cc rtlsdr.h fft.h buffer.h image.h
-	g++ $(FLAGS) $(GPU_FLAGS) -o gsm_scan gsm_scan.cc $(GPU_SRC) $(LIBS) -lrtlsdr -lfftw3 -lfftw3f
-ifdef USE_RPI_GPU_FFT
-	sudo chown root gsm_scan
-	sudo chmod a+s gsm_scan
-endif
+ogn-rf: src/ogn-rf.cc src/thread.h src/rtlsdr.h src/fft.h src/buffer.h src/image.h
+	g++ $(FLAGS) $(GPU_FLAGS) -o $@ $< $(GPU_SRC) $(LDLIBS)
+
+gsm_scan: src/gsm_scan.cc src/fft.h src/buffer.h src/image.h
+	g++ $(FLAGS) $(GPU_FLAGS) -o $@ $< $(GPU_SRC) $(LDLIBS)
+
+r2fft_test:	src/r2fft_test.cc src/r2fft.h src/fft.h
+	g++ $(FLAGS) $(GPU_FLAGS) -o $@ $< $(GPU_SRC) $(LDLIBS)
 
-r2fft_test:	Makefile r2fft_test.cc r2fft.h fft.h
-	g++ $(FLAGS) -o r2fft_test r2fft_test.cc -lm -lfftw3 -lfftw3f
+clean:
+	$(RM) gsm_scan ogn-rf r2fft_test
 
+.PHONY: clean
