From: "Fabian P. Schmidt" <kerel-fs@gmx.de>
Date: Sun, 12 Feb 2017 18:03:14 +0100
Subject: Makefile: Add all to phony targets

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 859ea8f..a9d2e41 100644
--- a/Makefile
+++ b/Makefile
@@ -35,4 +35,4 @@ r2fft_test:	src/r2fft_test.cc src/r2fft.h src/fft.h
 clean:
 	$(RM) gsm_scan ogn-rf r2fft_test
 
-.PHONY: clean
+.PHONY: all clean
